src/neptune_mlflow_plugin/impl/plugin.py

```diff
@@ -0,0 +1,291 @@
+                artifact_uri=self.store_uri,
+                run_name=run_name,
+            ),
+            run_data=RunData(),
+        )
+
+        self._run_provider.set_mlflow_run_id(run.info.run_id)
+
+        return run
+
+    def delete_run(self, run_id):
+        pass
+
+    def restore_run(self, run_id):
+        pass
+
+    def get_metric_history(self, run_id, metric_key, max_results=None, page_token=None):
+        pass
+
+    def _search_runs(self, experiment_ids, filter_string, run_view_type, max_results, order_by, page_token):
+        pass
+
+    def log_batch(self, run_id, metrics, params, tags):
+        for metric in metrics:
+            self.neptune_run[f"metrics/{metric.key}"].append(metric.value, step=metric.step, timestamp=metric.timestamp)
+
+        for param in params:
+            self.neptune_run[f"params/{param.key}"].append(param.value)
+
+    def log_inputs(self, run_id: str, datasets: Optional[List[DatasetInput]] = None):
+        pass
+
+    def record_logged_model(self, run_id, mlflow_model):
+        pass
+
+    def log_metric(self, run_id, metric: Metric):
+        self.neptune_run[metric.key].append(metric.value, step=metric.step, timestamp=metric.timestamp / 1e3)
+
+    def log_param(self, run_id, param: Param):
+        self.neptune_run[param.key] = param.value
+
+    def set_tag(self, run_id, tag):
+        if tag.key:
+            neptune_tags = [tag.key, tag.value]
+        else:
+            neptune_tags = tag.value
+
+        self.neptune_run["sys/tags"].add(neptune_tags)
+
+
+class NeptuneArtifactRepo(ArtifactRepository):
+    def __init__(self, *args, **kwargs):
+        super(NeptuneArtifactRepo, self).__init__(*args, **kwargs)
+        self._artifact_dir = local_file_uri_to_path(self.artifact_uri)
+        self._neptune_kwargs = parse_neptune_kwargs_from_uri(self.artifact_uri)
+
+        self._run_provider = RunProvider(self._neptune_kwargs)
+
+        self._api_token = self._neptune_kwargs.pop("api_token", None)
+        self._project = self._neptune_kwargs.pop("project", None)
+
+        self._neptune_run: Optional[Run] = None
+
+    @property
+    def neptune_run(self) -> Run:
+        if not self._neptune_run:
+            self._neptune_run = self._run_provider.get_or_create_neptune_run(self._run_provider.mlflow_run_id)
+
+        return self._neptune_run
+
+    def log_artifact(self, local_file, artifact_path=None):
+
+        target_path = artifact_path if artifact_path else Path(local_file).stem
+
+        self.neptune_run[target_path].upload(local_file)
+
+        self.neptune_run.sync()
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        target_path = artifact_path if artifact_path else Path(local_dir).stem
+
+        self.neptune_run[target_path].upload_files(os.path.join(local_dir, "*"))
+
+        self.neptune_run.sync()
+
+    def list_artifacts(self, path):
+        result = []
+
+        for entry in self.neptune_run[path].list_fileset_files():
```

Comment:
Try to use list comprehension as `append` looks weird and inefficient as we know the size of a target list:
```python
def list_artifacts(self, path):
    return [
        FileInfo(
            path=entry.name,
            is_dir=(entry.file_type == "directory"),
            file_size=entry.size
        )
        for entry in self.neptune_run[path].list_fileset_files()
    ]
```